include/sparrow/layout/primitive_array.hpp

```diff
@@ -155,6 +192,120 @@ namespace sparrow
         SPARROW_ASSERT_TRUE(get_arrow_proxy().data_type() == arrow_traits<T>::type_id);
     }
 
+    template <class T>
+    template <std::ranges::input_range VALUE_RANGE, std::ranges::input_range BOOL_RANGE>
+    requires std::convertible_to<std::ranges::range_value_t<VALUE_RANGE>, T> &&
+             std::convertible_to<std::ranges::range_value_t<BOOL_RANGE>, bool>
+    arrow_proxy primitive_array<T>::create_proxy(VALUE_RANGE && values, BOOL_RANGE && is_non_null)
+    {
+
+        const auto range_size = std::ranges::size(values);
+
+        // buffers
+        auto bitmap_ptr = new std::uint8_t[(range_size + 7) / 8];
+        auto data_buffer_ptr = new std::uint8_t[range_size * sizeof(T)];
+
+        #ifdef __GNUC__
+        #    pragma GCC diagnostic push
+        #    pragma GCC diagnostic ignored "-Wcast-align"
+        #endif
+        auto data_ptr = reinterpret_cast<T*>(data_buffer_ptr);
+        #ifdef __GNUC__
+        #    pragma GCC diagnostic pop
+        #endif
+
+        // iterators to the inputs
+        auto value_iter = std::ranges::begin(values);
+        auto is_non_null_iter = std::ranges::begin(is_non_null);
+
+        // fill the buffers
+        for(std::size_t i=0; i < range_size; ++i)
+        {
+            const bool is_non_null_val = *is_non_null_iter;
+            if(is_non_null_val)
+            {
+                data_ptr[i] = static_cast<T>(*value_iter);
+                // set bit to 1
+                bitmap_ptr[i / 8] |= static_cast<std::uint8_t>(1 << (i % 8));
+            }
+            else
+            {
+                // set bit to 0
+                bitmap_ptr[i / 8] &= static_cast<std::uint8_t>(~(1 << (i % 8)));
```

Comment:
this does not work in the generic case.
An iterator might might not be allowed to be dereferenced in the case of a missing value

Reply:
> You should set the value anyway, in case the bitmap is changed later.

if someone changes the value of the bitmap, one has change the value anyhow

> In the case the user want to set to null a single element in 100 000 elements, he has to create a vector of 100 000 elements or a range which return false for this index ?

It does not need to be a vector at all, it can be any iterator. Second, there is an overload where you just pass the values (this then generates an iterator always returning true (I trust compilers do do some sort of optimization st. this will not be very bad)) and then set by hand change the bitmap later on.
Furthermore, the price of an N element bool-vector would be 1/8 of the size of the actual values in the worst case.

Reply:
> And if you set a missing value to non-missing you better change the value anyhow
I don't know the use cases, but there is no cons and only benefits to copy "null" values.

Reply:
> I don't know the use cases, but there is no cons and only benefits to copy "null" values.

Again, the value iterator might be undefined for the places where data is missing.
Assume you have `std::vector<sparrow::nullable<float>>` and the values are a transform iterator and also the missing-vs-non missing is a transform-iterator.

In that case you will get an exception if you try to access `.value()` at places where you don't have data

Reply:
You have to use the get() function in that case